user_tools/src/spark_rapids_pytools/rapids/rapids_tool.py

```diff
@@ -218,19 +239,26 @@ def _connect_to_execution_cluster(self):
         Connecting to execution cluster
         :return:
         """
-        if self.runs_on_cluster:
+        if self.requires_cluster_connection():
             self.logger.info('%s requires the execution cluster %s to be running. '
                              'Establishing connection to cluster',
                              self.pretty_name(),
                              self.cluster)
             exec_cluster = self.ctxt.platform.connect_cluster_by_name(self.cluster)
+            self.ctxt.set_ctxt('execCluster', exec_cluster)
```

Comment:
Wondering if this should be after checking if the cluster is running. i.e after `self._verify_exec_cluster()` .
Also, line 250-252 seems redundant. We are already checking it in verify_exec_cluster().